### PR TITLE
qa/tests: try using `cephadm`

### PR DESCRIPTION
**Pull request by yuriw**
Signed-off-by: Yuri Weinstein <yweinste@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yuriw** [comment]
http://pulpito.ceph.com/teuthology-2020-06-17_17:24:17-rados:basic-master-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
```
2020-06-17T17:50:38.789 INFO:teuthology.orchestra.run.smithi117.stderr:Error ENOENT: ('name %s already in use', 'b')
debug 2020-06-17T17:50:38.782+0000 7f8276554700 -1 log_channel(cephadm) log [ERR] : _Promise failed
Traceback (most recent call last):
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 275, in _finalize
    next_result = self._on_complete(self._value)
  File "/usr/share/ceph/mgr/cephadm/module.py", line 131, in <lambda>
    return CephadmCompletion(on_complete=lambda _: f(*args, **kwargs))
  File "/usr/share/ceph/mgr/cephadm/module.py", line 2004, in add_mon
    return self._add_daemon('mon', spec, self.mon_service.create)
  File "/usr/share/ceph/mgr/cephadm/module.py", line 1957, in _add_daemon
    create_func, config_func)
  File "/usr/share/ceph/mgr/cephadm/module.py", line 1973, in _create_daemons
    forcename=name)
  File "/usr/share/ceph/mgr/cephadm/module.py", line 570, in get_unique_name
    raise orchestrator.OrchestratorValidationError('name %s already in use', forcename)
orchestrator._interface.OrchestratorValidationError: ('name %s already in use', 'b')
debug 2020-06-17T17:50:38.782+0000 7f8276554700 -1 mgr.server reply reply (2) No such file or directory ('name %s already in use', 'b')
2020-06-17T17:50:39.397 DEBUG:teuthology.orchestra.run:got remote process result: 2
2020-06-17T17:50:39.398 ERROR:teuthology.contextutil:Saw exception from nested tasks
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 32, in nested
    vars.append(enter())
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/github.com_yuriw_ceph_wip-yuriw-rados-master/qa/tasks/cephadm.py", line 502, in ceph_mons
    remote.shortname + ':' + ctx.ceph[cluster_name].mons[mon] + '=' + id_,
  File "/home/teuthworker/src/github.com_yuriw_ceph_wip-yuriw-rados-master/qa/tasks/cephadm.py", line 47, in _shell
    **kwargs
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 206, in run
    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 475, in run
    r.wait()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 162, in wait
    self._raise_for_status()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/run.py", line 184, in _raise_for_status
    node=self.hostname, label=self.label
teuthology.exceptions.CommandFailedError: Command failed on smithi117 with status 2: 'sudo /home/ubuntu/cephtest/cephadm --image quay.ceph.io/ceph-ci/ceph:9df7e03345a31351fa0be5602855e9fdcda5e993 shell -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyrin
```
One idea would be to use the qa/suites/rados/cephadm/smoke/fixed-2.yaml cluster.

**Comment by sebastian-philipp** [comment]
@yuriw any update on this?

**Comment by yuriw** [comment]
@sebastian-philipp not yet @joshd was reviewing

**Comment by yuriw** [comment]
testing on @jdurgin commit
http://pulpito.ceph.com/teuthology-2020-07-01_22:05:32-rados:basic-master-distro-basic-smithi/

**Comment by yuriw** [comment]
that ^ passed

and this is `cephadm`
https://pulpito.ceph.com/teuthology-2020-07-02_01:38:17-rados:cephadm-master-distro-basic-smithi

File: qa/tasks/cephadm.py [hunk 1]
```diff
@@ -364,11 +364,12 @@ def ceph_bootstrap(ctx, config, registry):
             '--output-keyring',
             '/etc/ceph/{}.client.admin.keyring'.format(cluster_name),
             '--output-pub-ssh-key', '{}/{}.pub'.format(testdir, cluster_name),
+            '--mon-id', first_mon,
+            '--mgr-id', first_mgr,
```
Comment (tchaikov) [inline qa/tasks/cephadm.py:368]:
```
2020-07-11T05:43:03.562 INFO:tasks.cephadm:Teardown complete
2020-07-11T05:43:03.562 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 368, in ceph_bootstrap
    '--mgr-id', first_mgr,
UnboundLocalError: local variable 'first_mgr' referenced before assignment

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/run_tasks.py", line 93, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 1217, in task
    lambda: cephfs_setup(ctx=ctx, config=config),
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 55, in nested
    raise exc[1]
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 1095, in initialize_config
    yield
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/contextutil.py", line 31, in nested
    vars.append(enter())
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-kefu-testing-2020-07-11-1109/qa/tasks/cephadm.py", line 471, in ceph_bootstrap
    ctx.daemons.get_daemon(type_, id_, cluster).stop()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/daemon/cephadmunit.py", line 149, in stop
    self.remote.sh(self.stop_cmd)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 248, in sh
    proc = self.run(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/teuthology/orchestra/remote.py", line 204, in run
    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
```

https://pulpito.ceph.com/kchai-2020-07-11_05:07:55-rados-wip-kefu-testing-2020-07-11-1109-distro-basic-smithi/

Comment (sebastian-philipp) [reply qa/tasks/cephadm.py:368]: hm. `--mon-id` and `--mgr-id` only make sense in roleless mode. We#re leaving the roleless mode when we require the `-mon-id` param. 

I'd say either we have to use the non-roleless mode or we have to fix the tests.

**Comment by sebastian-philipp** [comment]
unstale.

File: qa/tasks/cephadm.py (hunk 1 above)
Comment (sebastian-philipp) [reply qa/tasks/cephadm.py:368]: ok this change doesn't really make sense. can you revert this mon-ip thingy, please?

**Pull request closed without merging**